nip47/event_handler_test.go

```diff
@@ -558,6 +556,82 @@ func doTestHandleResponse_IncorrectEncryption(t *testing.T, svc *tests.TestServi
 	msg, err := cipher.Encrypt(string(payloadBytes))
 	assert.NoError(t, err)
 
+	// don't pass correct encryption
+	reqEvent := &nostr.Event{
+		Kind:      models.REQUEST_KIND,
+		PubKey:    reqPubkey,
+		CreatedAt: nostr.Now(),
+		Tags:      nostr.Tags{[]string{"encryption", requestEncryptionTag}},
+		Content:   msg,
+	}
+
+	err = reqEvent.Sign(reqPrivateKey)
+	assert.NoError(t, err)
+
+	relay := tests.NewMockRelay()
+
+	nip47svc.HandleEvent(context.TODO(), relay, reqEvent, svc.LNClient)
+
+	assert.NotNil(t, relay.PublishedEvents)
+	responseContent := relay.PublishedEvents[0].Content
+	msg, err = cipher.Decrypt(responseContent)
+	assert.NoError(t, err)
+	assert.NotEqual(t, "", msg)
+
+	unmarshalledResponse := models.Response{}
+
+	err = json.Unmarshal([]byte(msg), &unmarshalledResponse)
+	assert.NoError(t, err)
+	assert.Nil(t, unmarshalledResponse.Result)
+	// assert.Equal(t, models.GET_INFO_METHOD, unmarshalledResponse.ResultType)
+	assert.Equal(t, constants.ERROR_UNSUPPORTED_ENCRYPTION, unmarshalledResponse.Error.Code)
+	assert.Contains(t, unmarshalledResponse.Error.Message, "invalid encryption:")
+}
+
+func TestHandleResponse_EncryptionTagDoesNotMatchPayload(t *testing.T) {
+	svc, err := tests.CreateTestService(t)
+	require.NoError(t, err)
+	defer svc.Remove()
+	// encryption specifies what cipher will use. If constants.ENCRYPTION_TYPE_NIP44_V2 is passed,
+	// cipher must be NIP-44, otherwise cipher MUST be NIP-04
+	doTestHandleResponse_EncryptionTagDoesNotMatchPayload(t, svc, constants.ENCRYPTION_TYPE_NIP44_V2, constants.ENCRYPTION_TYPE_NIP04)
+	doTestHandleResponse_EncryptionTagDoesNotMatchPayload(t, svc, constants.ENCRYPTION_TYPE_NIP04, constants.ENCRYPTION_TYPE_NIP44_V2)
+	doTestHandleResponse_EncryptionTagDoesNotMatchPayload(t, svc, constants.ENCRYPTION_TYPE_NIP44_V2, "")
+}
+
+func doTestHandleResponse_EncryptionTagDoesNotMatchPayload(t *testing.T, svc *tests.TestService, requestEncryption, requestEncryptionTag string) {
+	albyOAuthSvc := alby.NewAlbyOAuthService(svc.DB, svc.Cfg, svc.Keys, svc.EventPublisher)
+	nip47svc := NewNip47Service(svc.DB, svc.Cfg, svc.Keys, svc.EventPublisher, albyOAuthSvc)
+
+	reqPrivateKey := nostr.GeneratePrivateKey()
+	reqPubkey, err := nostr.GetPublicKey(reqPrivateKey)
+	assert.NoError(t, err)
+
+	app, _, err := tests.CreateAppWithPrivateKey(svc, reqPrivateKey, constants.ENCRYPTION_TYPE_NIP44_V2)
+	assert.NoError(t, err)
+
+	appPermission := &db.AppPermission{
+		AppId: app.ID,
+		App:   *app,
+		Scope: constants.GET_BALANCE_SCOPE,
+	}
+	err = svc.DB.Create(appPermission).Error
+	assert.NoError(t, err)
+
+	content := map[string]interface{}{
+		"method": models.GET_INFO_METHOD,
+	}
+
+	payloadBytes, err := json.Marshal(content)
+	assert.NoError(t, err)
+
+	reqCipher, err := cipher.NewNip47Cipher(requestEncryption, *app.WalletPubkey, reqPrivateKey)
+	assert.NoError(t, err)
+	nip44Cipher, err := cipher.NewNip47Cipher(constants.ENCRYPTION_TYPE_NIP44_V2, *app.WalletPubkey, reqPrivateKey)
```

Comment:
Can we add a comment for readability saying that in such cases we always encrypt the response with `NIP-44`?

Reply:
added